rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -584,9 +594,46 @@ SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i)[\s'\"`()]*?\b([\d\w]+)\b[\s'\"`()]*?(?:
     ver:'OWASP_CRS/3.4.0-dev',\
     severity:'CRITICAL',\
     multiMatch,\
-    setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
-    setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
+    setvar:'tx.lhs_942130=%{TX.1}',\
+    chain"
+    SecRule MATCHED_VAR "@rx (?i)(?:(?:r(?:egexp|like)|<=>|=)[\s'\"`()]*?(?:\b%{tx.lhs_942130}\b)|(?:sounds\s+)?like[\s'\"`()]*?(?:\b%?%{tx.lhs_942130}%?\b))" \
+        "setvar:'tx.sql_injection_score=+%{tx.critical_anomaly_score}',\
+        setvar:'tx.anomaly_score_pl2=+%{tx.critical_anomaly_score}'"
 
+# Rule Targeting logical inequalities that return TRUE (e.g. 1 != 2)
+#
+# The regexp written at the initial rule on the chain is the original prefix from the 942130.data.
+# When the prefix matches, we store the match to a new variable for comparing in the chained rule.
+#
+# The chained rule then uses the assembled regexp from 942130.data to check for the right hand side of the logical query.
+# To rebuild the regexp:
+#   cd util/regexp-assemble
+#   ./regexp-assemble.py data/942131.data
```

Comment:
This comment needs to be updated. The regular expression for the starter rule can't be generated (at the moment).